functions/Backup-DbaDatabase.ps1

```diff
@@ -71,27 +75,47 @@ Backs up AdventureWorks2014 to sql2016's C:\temp folder
 #>
 	[CmdletBinding(DefaultParameterSetName = "Default")]
 	param (
-		[parameter(ParameterSetName = "NoPipe", Mandatory = $true)]
+		[parameter(ParameterSetName = "NoPipe", Mandatory = $true)][parameter(ParameterSetName = "Pipe")]
 		[object[]]$SqlInstance,
+		[parameter(ParameterSetName = "NoPipe")][parameter(ParameterSetName = "Pipe")]
 		[System.Management.Automation.PSCredential]$SqlCredential,
-		[string]$BackupDirectory,
+		[parameter(ParameterSetName = "NoPipe")][parameter(ParameterSetName = "Pipe")]
+		[string[]]$BackupDirectory,
+		[parameter(ParameterSetName = "NoPipe")][parameter(ParameterSetName = "Pipe")]
 		[string]$BackupFileName,
+		[parameter(ParameterSetName = "NoPipe")][parameter(ParameterSetName = "Pipe")]
 		[switch]$NoCopyOnly,
+		[parameter(ParameterSetName = "NoPipe")][parameter(ParameterSetName = "Pipe")]
 		[ValidateSet('Full', 'Log', 'Differential', 'Diff', 'Database')]
 		[string]$Type = "Database",
-		[parameter(ParameterSetName = "Pipe", Mandatory = $true, ValueFromPipeline = $true)]
-		[object[]]$DatabaseCollection
+		[parameter(ParameterSetName = "Pipe",Mandatory = $true, ValueFromPipeline = $true)]
+		[object[]]$Arrdatabases,
+		[parameter(ParameterSetName = "NoPipe")][parameter(ParameterSetName = "Pipe")]
+		[switch]$CreateFolder,
+		[parameter(ParameterSetName = "NoPipe")][parameter(ParameterSetName = "Pipe")]
+		[int]$FileCount=0
+
 		
 	)
 	DynamicParam { if ($SqlInstance) { return Get-ParamSqlDatabases -SqlServer $SqlInstance[0] -SqlCredential $SqlCredential } }
-	
 	BEGIN
 	{
 		$FunctionName = $FunctionName = (Get-PSCallstack)[0].Command
-				
+		$PipeDbs = @()
+		
+
+
 		if ($SqlInstance.length -ne 0)
 		{
-			$databases = $psboundparameters.Databases
+			if ($PSCmdlet.ParameterSetName -ne 'Pipe')
+			{	
+				Write-Verbose "$Functionname - Databases paramter used"
```

Comment:
typo here as well and hey! I didnt know you could get $pscmdlet.parametersetname